### DISCUSSION TITLE
Import forcing from Arakawa type simulation, definition of coastline

### DISCUSSION THREAD

**Discussion by samfredriksson** [OpenDrift/opendrift#931] [Q&A] [answered]
Hi,
I´m working with a quite detailed coast line within an archipelago where the detailed bathymetry and coast line becomes import. Our forcing data comes from an NEMO3.6 output using Arakawa-C grid (z*). Here the temperature, salinity are in T-position, whereas u, v and w are in U,V, and W, respectively. 
•	What is the best practice (tutorial etc.) to import this kind of forcing to Opendrift. Can it be imported as separate files interpolated within opendrift or should it be interpolated in advance to say T-position?
•	How is the coastline interaction defined? Is the coastline a polygon connecting the T-positions given in a mask file or is it related to the U, V, and W-positions in each file (depending on the first question) or even the cell extension as implied by the mask file? That is, can a stranded element be outside or not the T-positions if the cell extends outside the polygons defined by the T-points? 
//Sam

**Comment by knutfrode** [comment, marked as answer]
Hi Sam,

If your data was converted to a common, regular grid with z-coordinates in the vertical and complying to the CF-convention, the generic netCDF reader (reader_netCDF_CF_generic) could be used directly.

However, if would of course be a great advantage for NEMO users if the raw output files could be used without such post-processing. So a dedicated "NEMO-reader" would be welcome, but no one have made that so far (at least not for the main repository).
I guess this would be a little similar to the existing ROMS-reader, where variables are stored on various staggered grids, and with sigma-coordinates in the vertical. This reader performs the de-staggering and returns regular "blocks" of data with z in the vertical.
https://github.com/OpenDrift/opendrift/blob/master/opendrift/readers/reader_ROMS_native.py
I assume it should not be a huge task to make a NEMO-reader, starting from e.g. the ROMS reader. (However, I will not have time for this in the near future, and do also normally not deal with raw NEMO files).

Most often, the vector GSHHG coastline is used, but you may also use a raster variable from e.g. an ocean model output file, as demonstrated in this example:
https://opendrift.github.io/gallery/example_model_landmask.html
So an eventual "NEMO-reader" may define the landmask as found logical. The only requirement is that the variable must have an attribute `standard_name` equal to `land_binary_mask` (0=ocean, 1=land).

**Reply by samfredriksson** [reply]
Thanks
